File: c_plus/Reverse_Bits.cpp
#include<iostream>
using namespace std;

class Solution {
public:
    uint32_t reverseBits(uint32_t n) {
        uint32_t ret = 0;
        for (int i = 0; i < 32; i++) {
            ret = (ret * 2) + (n & 0x1);
            n /= 2;
        }
        return ret;
    }
};

int main() {
    Solution s;
    uint32_t t = 43261596;
    cout << s.reverseBits(t)  << endl;
}

File: py/Reverse_Bits.py
class Solution:
    def reverseBits(self, nums):
        b = bin(nums)[: 1 : -1]
        return int(b + '0' * (32 - len(b)), 2)

if __name__ == '__main__':
    s = Solution()
    print(s.reverseBits(43261596))
    # s = [1, 2, 3, 4, 5]
    # print(s[: 1 : -1])